src/mono/mono/mini/mini-arm64.c

```diff
@@ -3966,7 +3969,7 @@ mono_arch_output_basic_block (MonoCompile *cfg, MonoBasicBlock *bb)
 		case OP_EXPAND_R4:
 		case OP_EXPAND_R8: {
 			const int t = get_type_size_macro (ins->inst_c1);
-			arm_neon_fdup_e (code, VREG_FULL, t, dreg, sreg1, 0);
+			arm_neon_fdup_e (code, get_vector_size_macro (ins), t, dreg, sreg1, 0);
```

Comment:
Most arm64 vector instructions are not defined when working on a 64bit vector of 64bit elements (i.e. one value in a vector). You will get an invalid instruction error. A 64b broadcast of a 64bit value is merely a `mov` if `dreg != sreg1`.

Reply:
Do you propose that we keep only the 128bit version even for 64bit vector? When I was testing this I actually did get `dup.2s` instruction not the `dup.4s`.

Reply:
I checked against documentation and `Vector64<double>`, `Vector64<long>` seem allowed. If we are to support these on arm64, we will need to have a separate codepath, as the vector instructions do not specialize to such "vectors". These will mostly materialize as scalar FP instructions.

Reply:
Ok, I'll make a change so that `OP_EXPAND_R8` uses the original `VREG_FULL` and `OP_EXPAND_R4` keeps the new change (as such we will have `Vector2` support). I'll leave the actual support for `OP_EXPAND_R8` for `Vector64<double>` to future work when enabling `Vector64` on arm64 JIT-mini.